### PR TITLE
Allow building with GHC 9.2

### PR DESCRIPTION
**Pull request by jhrcek**
I'd love to be able to compile a project of mine (which has co-log as dependency) with GHC 9.2.4.

Could you please accept this PR bumping base version bound for that purpose?
I also extended CI config and fixed 2 warnings issues by the new version of ghc..

### CODE REVIEW

**Comment by alaendle** [comment]
Many thanks - LGTM 👍

**Comment by jhrcek** [comment]
Hello @vrom911 could you please check this PR and merge if it looks ok?
I also wanted to make it buildable with ghc 9.4, but that is blocked by https://github.com/co-log/co-log-core/pull/22/ ..

**Comment by jhrcek** [comment]
Hello @vrom911 
Could you please check this PR and merge if you get some time?
It would be great to have 9.2.4 compliant version on hackage :pray:

**Comment by jhrcek** [comment]
Hello @chshersh 
could you please have a look at this PR as one of the co-maintainers :blush: ?

**Review by vrom911 — APPROVED** [review]
Thanks!

**Pull request merged**
